servers/Azure.Mcp.Server/tests/Azure.Mcp.Server.UnitTests/Infrastructure/CommandMetadataSyncTests.cs

```diff
@@ -52,45 +53,56 @@ public void AzCommandsMetadata_Should_Be_Synchronized()
             "This ensures that tool metadata in the documentation stays synchronized with the actual command implementations.");
     }
 
-    private static (int ExitCode, string Output, string Error) RunCommand(string command, string arguments, string workingDirectory)
+    private static string FindPowerShellExecutable()
     {
-        var processInfo = new ProcessStartInfo
-        {
-            FileName = command,
-            Arguments = arguments,
-            WorkingDirectory = workingDirectory,
-            RedirectStandardOutput = true,
-            RedirectStandardError = true,
-            UseShellExecute = false,
-            CreateNoWindow = true
-        };
-
-        using var process = new Process { StartInfo = processInfo };
-        var outputBuilder = new System.Text.StringBuilder();
-        var errorBuilder = new System.Text.StringBuilder();
-
-        process.OutputDataReceived += (sender, args) =>
+        // Try common PowerShell paths
+        var commonPaths = OperatingSystem.IsWindows()
+            ? new[]
+            {
+                Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), "PowerShell", "7", "pwsh.exe"),
+                Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), "PowerShell", "pwsh.exe"),
+                "C:\\Program Files\\PowerShell\\7\\pwsh.exe",
+                "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe"
+            }
+            : new[] { "/usr/local/bin/pwsh", "/usr/bin/pwsh" };
+
+        foreach (var path in commonPaths)
         {
-            if (args.Data != null)
+            if (File.Exists(path))
             {
-                outputBuilder.AppendLine(args.Data);
+                return path;
             }
-        };
+        }
 
-        process.ErrorDataReceived += (sender, args) =>
+        // Search in PATH environment variable
+        var pathEnv = Environment.GetEnvironmentVariable("PATH");
+        if (pathEnv != null)
         {
-            if (args.Data != null)
+            var exeName = OperatingSystem.IsWindows() ? "pwsh.exe" : "pwsh";
+            foreach (var directory in pathEnv.Split(Path.PathSeparator))
             {
-                errorBuilder.AppendLine(args.Data);
+                var fullPath = Path.Combine(directory, exeName);
+                if (File.Exists(fullPath))
+                {
+                    return fullPath;
+                }
             }
-        };
 
-        process.Start();
-        process.BeginOutputReadLine();
-        process.BeginErrorReadLine();
-        process.WaitForExit();
+            // Also try powershell.exe on Windows as fallback
+            if (OperatingSystem.IsWindows())
```

Comment:
You almost never should use the old built in powershell.exe so I would remove this.